dashboard/pages/1_comparing topics between papers.py

```diff
@@ -0,0 +1,38 @@
+import os
+from dotenv import load_dotenv
+import streamlit as st
+import psycopg2
+import pandas as pd
+import plotly.express as px
+
+load_dotenv()
+
+
+def connect_to_database() -> psycopg2.extensions.connection:
+    """Connects to the PostgreSQL database"""
+    try:
+        conn = psycopg2.connect(
+            host=os.getenv('DB_HOST'),
+            port=os.getenv('DB_PORT'),
+            database=os.getenv('DB_NAME'),
+            user=os.getenv('DB_USERNAME'),
+            password=os.getenv('DB_PASSWORD')
+        )
+        return conn
+    except Exception as e:
+        st.error(f"Database connection failed: {e}")
+        raise
+
+
+@st.cache_data()
```

Comment:
nice caching!

Reply:
Mabye here you would want to specify how long you want the cache for. So for 10 mins the code would be 
```suggestion
@st.cache_data(ttl=600)
```
See https://docs.streamlit.io/develop/api-reference/caching-and-state/st.cache_data for more.